client/rpc/src/state/mod.rs

```diff
@@ -347,6 +357,19 @@ impl<Block, Client> StateApi<Block::Hash> for State<Block, Client>
 	) -> RpcResult<bool> {
 		self.backend.unsubscribe_runtime_version(meta, id)
 	}
+
+	/// Re-execute the given block with the tracing targets given in `targets`
+	/// and capture all state changes.
+	///
+	/// Note: requires the node to run with `--rpc-methods=Unsafe`.
+	fn trace_block(&self, block: Block::Hash, targets: Option<String>) -> FutureResult<sp_rpc::tracing::BlockTrace> {
+		if let Err(err) = self.deny_unsafe.check_if_safe() {
+			// TODO: there has to be a more elegant way to do this.
+			return Box::new(async move { Err(err.into()) }.boxed().compat());
```

Comment:
@bkchr this is terrible; what can I do better here? :/

It's a bit odd that different RPCs use different return types, for `system` there's `bail_if_unsafe!` but that doesn't work here.

Reply:
Something like:
`return Box::new(rpc::futures::future::result(Err(err))`?

Reply:
ofc, ty very much.